library/src/main/java/com/crossbowffs/remotepreferences/RemotePreferenceProvider.java

```diff
@@ -319,15 +387,31 @@ public int update(Uri uri, ContentValues values, String selection, String[] sele
      */
     @Override
     public void onSharedPreferenceChanged(SharedPreferences prefs, String prefKey) {
-        ContentResolver resolver = getContext().getContentResolver();
+        String prefName = null;
+        boolean isDeviceProtected = false;
+        Context context = getContext();
+
         for (Map.Entry<String, SharedPreferences> entry : mPreferences.entrySet()) {
             if (entry.getValue() == prefs) {
-                String prefName = entry.getKey();
-                Uri uri = getPreferenceUri(prefName, prefKey);
-                resolver.notifyChange(uri, null);
-                return;
+                prefName = entry.getKey();
```

Comment:
Let's refactor this and the following loops into a new method `private RemotePreferenceFile getFileForPreferences(SharedPreferences prefs)`. Should make the code a bit easier to read.